r/R/RecordBatch.R

```diff
@@ -98,6 +98,7 @@
 #' @export
 record_batch <- function(..., schema = NULL){
   arrays <- list2(...)
+  # making sure there are always names
```

Comment:
Right but why? IIUC this has to do with how the cpp code distinguishes things to autosplice, switching behavior on `nchar(name)`. That's a subtlety I'll probably forget by Monday so it seems worth explaining.

Reply:
``` r
names(list(a = 1, b = 2))
#> [1] "a" "b"
names(list(a = 1, 2))
#> [1] "a" ""
names(list(1, 2))
#> NULL
```

otherwise we'd have to treat the NULL case differently internally as we did before, maybe I could do that instead.